src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1232,11 +1233,12 @@ public void run() {
                         } else if (e instanceof RWServerFoundException) {
                             LOG.info(e.getMessage());
                         } else {
+                            SocketAddress remoteAddr = clientCnxnSocket.getRemoteSocketAddress();
                             LOG.warn(
                                     "Session 0x"
                                             + Long.toHexString(getSessionId())
                                             + " for server "
-                                            + clientCnxnSocket.getRemoteSocketAddress()
+                                            + (remoteAddr == null ? addr : remoteAddr)
```

Comment:
I believe that, if `clientCnxnSocket.getRemoteSocketAddress()` is non-null, the value is always the same as `addr`.

If so, then this could be simplified to just `addr` without the ternary operation.

Reply:
Of course this comment is [non-blocking]

Reply:
Makes sense. I'll just use 'addr' here.